### PR TITLE
docs(messaging): update web example in line with flutter 2.2.0 generated `index.html`

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

Update messaging web docs example in line with latest stable flutter version `index.html`. direct users to example for setup.

## Related Issues

closes https://github.com/FirebaseExtended/flutterfire/issues/6372

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_messaging/firebase_messaging/example/web/firebase-messaging-sw.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ firebase.initializeApp({
   messagingSenderId: "448618578101",
   appId: "1:448618578101:web:ecaffe2bc4511738",
 });
-
+//// Necessary to receive background messages:
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging/example/web/firebase-messaging-sw.js:13]:
```suggestion
// Necessary to receive background messages:
```

File: packages/firebase_messaging/firebase_messaging/example/web/firebase-messaging-sw.js [hunk 2]
```diff
@@ -10,10 +10,10 @@ firebase.initializeApp({
...
 const messaging = firebase.messaging();
 
-// Optional:
+//// Optional:
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging/example/web/firebase-messaging-sw.js:16]:
```suggestion
// Optional:
```

File: packages/firebase_messaging/firebase_messaging/example/web/index.html [hunk 3]
```diff
@@ -24,11 +24,70 @@
   </script>
   <script src="main.dart.js" type="application/javascript"></script>
   <script>
-    if ("serviceWorker" in navigator) {
-      window.addEventListener("load", function () {
-        navigator.serviceWorker.register("/flutter_service_worker.js");
+     var serviceWorkerVersion = null;
```
Comment (rrousselGit) [inline packages/firebase_messaging/firebase_messaging/example/web/index.html:27]:
```suggestion
    var serviceWorkerVersion = null;
```

**Review by rrousselGit — APPROVED** [review]

**Review by Salakar — APPROVED** [review]

**Comment by JohnGalt1717** [comment]
It seems to me that if there is a way to register the configuration once and share it between the service worker and the main initialization that would be good. However, I tried defining a variable with it and using it in both places, but it didn't work. Is is possible with the service worker to do this? It would make the code way less confusing and avoid having to update the app in 2 places with the same information.

**Comment by JohnGalt1717** [comment]
It seems to me that if there is a way to register the configuration once and share it between the service worker and the main initialization that would be good. However, I tried defining a variable with it and using it in both places, but it didn't work. Is is possible with the service worker to do this? It would make the code way less confusing and avoid having to update the app in 2 places with the same information.

**Comment by russellwheatley** [comment]
@JohnGalt1717, it is on our radar to have a way to initialise your app with config from your app code (dart). I'm not able to show you how to share config between the service worker and the main initialisation without putting time aside to look into it. Where there's a will there's a way. If you do figure it out, we happily accept PRs that improve the code base 😄

**Comment by JohnGalt1717** [comment]
So I made it work:

Create a firebase-config.js file in the /web folder, put in the var firebaseConfig = {} in there like this:

```
var firebaseConfig = {
...
        };

firebase.initializeApp(firebaseConfig);

if ('serviceWorker' in navigator) {
    // Service workers are supported. Use them.
    window.addEventListener('load', function () {
        navigator.serviceWorker.register("/firebase-messaging-sw.js");

    });
}

```

Add the script tag immediately below the firebase script tags in your index.html like this:

```
    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-app.js"></script>
    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-messaging.js"></script>
    <script src="firebase-config.js"></script>

```

In firebase-messaging-sw.js change to:

```
importScripts("https://www.gstatic.com/firebasejs/8.6.1/firebase-app.js");
importScripts("https://www.gstatic.com/firebasejs/8.6.1/firebase-messaging.js");
importScripts("firebase-config.js");

const messaging = firebase.messaging();

// Optional:
// messaging.onBackgroundMessage((message) => {
//   console.log("onBackgroundMessage", message);
// });

```

No further changes are required, firebase is entirely encapsulated 3 script tags and 2 javscript files that don't interact with any of the flutter initialization code itself and thus eliminates all of the errors that that is prone to, and there's no repetition of configuration either.

**Comment by JohnGalt1717** [comment]
PS if you're talking to the flutter team, it would be really nice if when we write a library we could have the ability to include a web folder in the plugin that would result in files being copied into the destination project, and a way of templating index.html insertions for that code. There are a lot of plugins that need to do this (OpenIdConnect, HLS support for video, etc.)

**Pull request merged**